apps/mac.c

```diff
@@ -48,6 +52,27 @@ const OPTIONS mac_options[] = {
     {NULL}
 };
 
+static char *alloc_mac_algorithm_name(STACK_OF(OPENSSL_STRING) **optp,
+                                      const char *name, const char *arg)
+{
+    size_t len = strlen(name) + strlen(arg) + 2;
+    char *res = app_malloc(len, "algorithm name");
```

Comment:
oops - double malloc..

Reply:
I just spent the last half hour trying to locate this and was about to give up :(

Reply:
Most people do double frees - you are special...